<!--
   Run the fiducial_slam node
-->
<launch>
  <arg name="mapping_mode" default="false"/>
  <arg name="camera_frame"/>
  <arg name="map_frame"/>
  <arg name="odom_frame"/>
  <arg name="pose_frame"/>
  <arg name="publish_tf" default="true"/>


  <node type="fiducial_slam.py" pkg="fiducial_slam" output="screen" name="fiducial_slam">
    <param name="map_file" value="$(env HOME)/.ros/slam/map.txt" />
    <param name="obs_file" value="$(env HOME)/.ros/slam/obs.txt" />
    <param name="trans_file" value="$(env HOME)/.ros/slam/trans.txt" />
    <param name="mapping_mode" value="$(arg mapping_mode)" />
    <param name="camera_frame" value="$(arg camera_frame)" />
    <param name="map_frame" value="$(arg map_frame)" />
    <param name="odom_frame" value="$(arg odom_frame)" />
    <param name="pose_frame" value="$(arg pose_frame)" />
    <param name="publish_tf" value="$(arg publish_tf)" />
    <remap to="/fiducial_detect/fiducial_transforms" from="/fiducial_transforms"/>
  </node>
</launch>
